### PR TITLE
Refactor browser banner test file to new format

### PR DESCRIPTION
**Pull request by SriHV**
[138](https://github.com/orgs/ONSdigital/projects/21/views/1?pane=issue&itemId=72216515)

Refactored the browser-banner macro.spec file as per the new format.

### How to review this PR

Ask yourself....
- Do we need any additional tests?
- Are the tests descriptions clear?
- Does the order of the test document make sense?
- Is it internally consistent?
- Is it consistent with the [guidelines for test refactoring](https://confluence.ons.gov.uk/display/SDC/Testing+Refactor+Guidance)? 

run this test command to view the test results:
`yarn test --testNamePattern="FOR: browser-banner"`

### Checklist

This needs to be completed by the person raising the PR.

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/browser-banner/_macro.spec.js [hunk 1]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
-            renderComponent('browser-banner', {
-                ...EXAMPLE_BROWSER_BANNER_DEFAULT,
-                wide: true,
-            }),
-        );
-
-        expect($('.ons-container').hasClass('ons-container--wide')).toBe(true);
+describe('FOR: browser-banner', () => {
+    describe('GIVEN: Params: default', () => {
+        describe('WHEN: params are at default and language is default', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:10]:
```suggestion
        describe('WHEN: params are at default state', () => {
```
Language is a param so its covered by the first part of the statement

File: src/components/browser-banner/_macro.spec.js [hunk 2]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+            const $ = cheerio.load(renderComponent('browser-banner', {}));
+
+            it('passes jest-axe checks with', async () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:13]:
```suggestion
            it('passes jest-axe checks', async () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 3]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                const results = await axe($.html());
+                expect(results).toHaveNoViolations();
+            });
+
+            it('has the english version of default `bannerLeadingText`', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:18]:
```suggestion
            it('has the default bannerLeadingText', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 4]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                const bannerLeadingText = $('.ons-browser-banner__lead').text().trim();
+                expect(bannerLeadingText).toBe('This website no longer supports your browser.');
+            });
+
+            it('has the english version of default `bannerCTA`', () => {
+                const bannerCtaHtml = $('.ons-browser-banner__cta').text().trim();
+                expect(bannerCtaHtml).toBe('You can upgrade your browser to the latest version.');
+            });
+
+            it('has the english version of default `bannerLinkUrl`', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:28]:
```suggestion
            it('has the default bannerLinkUrl', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 5]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                expect($('.ons-browser-banner__link').attr('href')).toBe('https://www.ons.gov.uk/help/browsers');
+            });
+        });
+
+        describe('WHEN: params are at default and language is set to welsh', () => {
+            const $ = cheerio.load(renderComponent('browser-banner', { lang: 'cy' }));
+
+            it('has the welsh version of default `bannerLeadingText`', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:36]:
```suggestion
            it('THEN: has the welsh default bannerLeadingText', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 6]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                const bannerLeadingText = $('.ons-browser-banner__lead').text().trim();
+                expect(bannerLeadingText).toBe('Nid yw’r wefan hon yn cefnogi eich porwr mwyach.');
+            });
+
+            it('has the welsh version of default `bannerCTA`', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:41]:
```suggestion
            it('THEN: has the welsh default bannerCTA', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 7]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                const bannerCtaHtml = $('.ons-browser-banner__cta').text().trim();
+                expect(bannerCtaHtml).toBe('Gallwch ddiweddaru eich porwr i’r fersiwn ddiweddaraf.');
+            });
+
+            it('has the welsh version of default `bannerLinkUrl`', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:46]:
```suggestion
            it('THEN: has the welsh default bannerLinkUrl', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 8]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
-        const $ = cheerio.load(
-            renderComponent('browser-banner', {
-                ...EXAMPLE_BROWSER_BANNER_DEFAULT,
-            }),
-        );
-
-        expect($('.ons-container').hasClass('ons-container--full-width')).toBe(false);
+    describe('GIVEN: Params: wide', () => {
+        describe('WHEN: wide is set to true', () => {
+            it('has `container--wide` class', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:54]:
```suggestion
            it('THEN: has container--wide class', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 9]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                        wide: true,
+                    }),
+                );
+
+                expect($('.ons-container').hasClass('ons-container--wide')).toBe(true);
+            });
+        });
+
+        describe('WHEN: wide is not set', () => {
+            it('does not have `container--wide` class', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:67]:
```suggestion
            it('THEN: does not have container--wide class', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 10]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
-        expect(bannerCtaHtml).toBe('Gallwch ddiweddaru eich porwr i’r fersiwn ddiweddaraf.');
-    });
-
-    it('has the welsh version of default `bannerLinkUrl`', () => {
-        const $ = cheerio.load(renderComponent('browser-banner', { lang: 'cy' }));
 
-        expect($('.ons-browser-banner__link').attr('href')).toBe('https://cy.ons.gov.uk/help/browsers');
+    describe('GIVEN: Params: fullWidth', () => {
+        describe('WHEN: fullWidth is set to true', () => {
+            it('has `container--full-width` class', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:81]:
```suggestion
            it('THEN: has container--full-width class', () => {
```

File: src/components/browser-banner/_macro.spec.js [hunk 11]
```diff
@@ -5,102 +5,101 @@ import * as cheerio from 'cheerio';
...
+                        fullWidth: true,
+                    }),
+                );
+
+                expect($('.ons-container').hasClass('ons-container--full-width')).toBe(true);
+            });
+        });
+
+        describe('WHEN: fullWidth is not set', () => {
+            it('does not have `container--full-width` class when `fullWidth` is not set', () => {
```
Comment (rmccar) [inline src/components/browser-banner/_macro.spec.js:94]:
```suggestion
            it('THEN: does not have container--full-width class', () => {
```

**Review by alessioventuriniAND — APPROVED** [review]

**Review by rmccar — APPROVED** [review]

**Pull request merged**